### PR TITLE
Uninstall embed, entity_embed

### LINKED ISSUE #5312: Can we remove `embed` module?
As part of trying to understand what was changing in #5311, I think I may have discovered that we no longer need the `embed` module. We used to use it on V2 sites as a dependency for `entity_embed`, which allowed `<drupal-entity>` tags to be transformed via filter to their respective entities. This was the old method for embedding media objects in WYSIWYG text areas, but we changed that in V3 when we built it and eventually updated V2 to match in this PR: https://github.com/uiowa/uiowa/pull/3048

If we are no longer using the module, we should remove it and it associated config footprint so that we don't have to update it for no reason in the future.

### PR DESCRIPTION
**Pull request by joewhitsitt**
Resolves #5312 

- remove entity embed
- remove embed

# To test

```
ddev blt ds --site=hr.uiowa.edu && ddev drush @hr.local uli node/add/page
```

- There are no import errors when syncing a v2 and a v3 website and the modules are not enabled. Had to import the text formats first before core.extension tries to uninstall the module.
- HR should still be using media library in wysiwyg which we switched out a long time ago. You should be able to select/place/render as usual.
- HR should still be able to use entity browser (featured content) to find and select items for display (no relation but can be confusing based on the name)
- The code that was removed is a hook that only applies to entity_embed and pre-dates the uiowa repo.
- ~~Should probably run a scan to see if there is any remaining `<drupal-entity...` markup still within v2 sites.~~ This stuff hasn't been rendering anyway because it is no longer allowed in filtered_html.

# Follow-up

- Remove from composer.

### CODE REVIEW

**Comment by joewhitsitt** [comment]
As a light scan, I searched for `drupal-entity` within HR's find text tool. No results. `drupal-media` returned a lot. Almost thought it was going to break.

**Comment by joewhitsitt** [comment]
i think i have some chicken/egg stuff going on. poking

**Comment by pyrello** [comment]
Pulling to test.

File: config/default/filter.format.full_html.yml [hunk 1]
```diff
@@ -147,3 +140,10 @@ filters:
         small__ultrawide: small__ultrawide
         small__widescreen: small__widescreen
       allowed_media_types: {  }
+  filter_iframe:
+    id: filter_iframe
+    provider: uiowa_core
+    status: false
+    weight: -40
+    settings:
+      allowed_sources: ''
```
Comment (pyrello) [inline config/default/filter.format.full_html.yml:149]: Wondering where this came from?

Comment (joewhitsitt) [reply config/default/filter.format.full_html.yml:149]: i think filter_iframe was added after the last time we exported filter.format.full_html and that all filters are registered with the format regardless of whether the status is true or false.

Comment (joewhitsitt) [reply config/default/filter.format.full_html.yml:149]: aka long overdue config update

**Comment by pyrello** [comment]
When I try to sync HR, I'm getting the following config status warning:
```
 -------------------------------------------- ----------- 
  Name                                         State      
 -------------------------------------------- ----------- 
  core.entity_form_display.node.page.default   Different  
 -------------------------------------------- ----------- 
```
So the sync "fails".

No issues syncing the default site.

**Comment by joewhitsitt** [comment]
I had gotten that at one point and thought it was a local env blip. It seemed related to the summary/body work and that it wasn't stacking sitenow_v2 properly.

**Comment by pyrello** [comment]
> I had gotten that at one point and thought it was a local env blip. It seemed related to the summary/body work and that it wasn't stacking sitenow_v2 properly.

It wasn't immediately clear to me how to resolve it.

**Comment by joewhitsitt** [comment]
Bummer. I can't reproduce this with `main`. Something in this PR is causing problems...

**Comment by joewhitsitt** [comment]
On `main` something is funky with sitenow_v2. Syncing a site, everything looks normal when you go to /node/add/page. Go to the entity form display though and body/simple sitemap are at the top and when you save, it makes /node/add/page show the body above the paragraphs stuff.

**Comment by joewhitsitt** [comment]
@cory-skeers do you recall how v2 is hiding/disabling the body field after the body/summary PR?

**Comment by cory-skeers** [comment]
> On `main` something is funky with sitenow_v2. Syncing a site, everything looks normal when you go to /node/add/page. Go to the entity form display though and body/simple sitemap are at the top and when you save, it makes /node/add/page show the body above the paragraphs stuff.

Hrm... I pushed up a small change [943b277](https://github.com/uiowa/uiowa/pull/6898/commits/943b277aafc798a5072867af9ca1cffc3ea44e8f) that at least got the sync working nicely for me, but less sure of it now. It might need to be redone if there's some other stuff going on.

**Comment by cory-skeers** [comment]
> @cory-skeers do you recall how v2 is hiding/disabling the body field after the body/summary PR?

https://github.com/uiowa/uiowa/blob/0fa84d6c53c4ceedc6fb84ea08114dc32b120aa5/docroot/modules/custom/sitenow_p2lb/sitenow_p2lb.module#L43C8-L74

**Comment by cory-skeers** [comment]
> > @cory-skeers do you recall how v2 is hiding/disabling the body field after the body/summary PR?
> 
> https://github.com/uiowa/uiowa/blob/0fa84d6c53c4ceedc6fb84ea08114dc32b120aa5/docroot/modules/custom/sitenow_p2lb/sitenow_p2lb.module#L43C8-L74

Wait, misread that...

**Comment by joewhitsitt** [comment]
I think we missed entity form display for v2 https://github.com/uiowa/uiowa/pull/6769/files

v2 needs to show the content_block field and hide the body.

**Comment by cory-skeers** [comment]
> I think we missed entity form display for v2 https://github.com/uiowa/uiowa/pull/6769/files
> 
> v2 needs to show the content_block field and hide the body.

For the teaser->summary work, the v2 node form should have ended up staying the same. Disabling the field teaser field happens in code, and doesn't happen if it's a v2 page.

I tried syncing hawkeyemarchingband, and its page form seems to be working as expected w/paragraphs. I am seeing the same weird stuff when syncing hr, though.

**Comment by joewhitsitt** [comment]
Not sure what I may have broke by splitting the page form for v2 here but this seems to resolve the HR issue for me and still maintains the related content and unit fields that are unique to their site.

**Comment by cory-skeers** [comment]
> Not sure what I may have broke by splitting the page form for v2 here but this seems to resolve the HR issue for me and still maintains the related content and unit fields that are unique to their site.

Looking at the files diff, it's only modifying the existing patch a bit, and looks like the changes are good. I think somewhere along the line here it got removed and re-added? Whatever the case, looks to work!

**Comment by pyrello** [comment]
> Not sure what I may have broke by splitting the page form for v2 here but this seems to resolve the HR issue for me and still maintains the related content and unit fields that are unique to their site.

When I look at this commit, it looks like you are adding the form config patch: https://github.com/uiowa/uiowa/pull/6898/commits/273466dd93c2ccd0c767cbb74b25cb9040d8825f

But when I look at the file changes, it looks like it is just updating that file:
https://github.com/uiowa/uiowa/pull/6898/files#diff-92cf7420b9756062c7aeb453307ea5098a02fbdfc42e2cd9e9daab77d75a01b9

So maybe an earlier commit accidentally removed it?

**Comment by pyrello** [comment]
> So maybe an earlier commit accidentally removed it?

Yep, it looks like it was removed in an earlier commit: https://github.com/uiowa/uiowa/pull/6898/files#diff-92cf7420b9756062c7aeb453307ea5098a02fbdfc42e2cd9e9daab77d75a01b9

**Comment by pyrello** [comment]
Going to test this again.

**Comment by pyrello** [comment]
No issues with syncing HR now, so that's good 😅 

I'm noticing the following error message when I visit the node edit form:
```
Error message
Warning: Undefined array key "weight" in Drupal\Core\Entity\Entity\EntityFormDisplay->processForm() (line 215 of core/lib/Drupal/Core/Entity/Entity/EntityFormDisplay.php).
```

I'm going to check if we are getting that error in `main`.

**Comment by pyrello** [comment]
> I'm going to check if we are getting that error in `main`.

This is also present in `main` so represent a regression we introduced somewhere besides this PR.

**Review by pyrello — APPROVED** [review]
Reviewed code changes, tested HR per instructions, and synced default and tested adding content with media embed. No issues I could see. Nice work on this, @joewhitsitt! I'm pumped to be getting this done!

**Pull request merged**